### PR TITLE
DR2-1709 Add child count to case classes

### PR DESCRIPTION
**Pull request by MancunianSam**
DR2-1709 Add childCount to Dynamo case classes

This allows us to read the childCount out of Dynamo. The ticket says that this should be on all rows so I've made it mandatory.

### CODE REVIEW

File: ingest-asset-opex-creator/src/test/scala/uk/gov/nationalarchives/ingestassetopexcreator/testUtils/ExternalServicesTestUtils.scala [hunk 1]
```diff
@@ -155,6 +155,9 @@ class ExternalServicesTestUtils(dynamoServer: WireMockServer, s3Server: WireMock
        |      "sortOrder": {
        |        "N": "1"
        |      },
+       |      "childCount": {
+       |        "N": "0"
```
Comment (techncl) [inline ingest-asset-opex-creator/src/test/scala/uk/gov/nationalarchives/ingestassetopexcreator/testUtils/ExternalServicesTestUtils.scala:159]: Just out of curiosity, is it possible to have 0 children? If not, should we do a check/prevent this from happening?

**Review by techncl — COMMENTED** [review]
Looks good! Just one question as well as this

Since `childCount` is a numeric field, do you think we should add the `invalidNumericField` and `invalidNumericValue` tests for it?

File: ingest-asset-opex-creator/src/test/scala/uk/gov/nationalarchives/ingestassetopexcreator/testUtils/ExternalServicesTestUtils.scala (hunk 1 above)
Comment (MancunianSam) [reply ingest-asset-opex-creator/src/test/scala/uk/gov/nationalarchives/ingestassetopexcreator/testUtils/ExternalServicesTestUtils.scala:159]: It's possible to have zero children. Files don't have any children.

**Comment by MancunianSam** [comment]
Good point about the `invalidNumericField` and `invalidNumericValue` tests. I've added them in.

**Pull request merged**